src/sqlite3gen.cpp

```diff
@@ -1463,14 +2526,29 @@ static sqlite3* openDbConnection()
   QDir sqlite3Dir(outputDirectory);
   sqlite3 *db;
   int rc;
+  struct stat buf;
 
   rc = sqlite3_initialize();
   if (rc != SQLITE_OK)
   {
     msg("sqlite3_initialize failed\n");
     return NULL;
   }
-  rc = sqlite3_open_v2(outputDirectory+"/doxygen_sqlite3.db", &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, 0);
+
+
+  if (stat (outputDirectory+"/doxygen_sqlite3.db", &buf) == 0)
+  {
+    msg("doxygen_sqlite3.db already exists! aborting sqlite3 output generation!\n");
+    msg("If you wish to re-generate the database, remove or archive the existing copy first.\n");
+    return NULL;
```

Comment:
Not sure if this is how you'd prefer I check file existence. I couldn't find a clear consensus approach and don't know enough to evaluate the tradeoffs between methods I found.